### PR TITLE
Convert govuk button to jinja macro

### PR DESCRIPTION
**Pull request by leohemsted**
the last and most-used macro. A few changes to how we query for it in tests (and once in code) but otherwise relatively straightforward.

- [x] https://github.com/alphagov/notifications-admin/pull/4429
- [x] https://github.com/alphagov/notifications-admin/pull/4434

### CODE REVIEW

File: tests/javascripts/preventDuplicateFormSubmissions.test.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ describe('Prevent duplicate form submissions', () => {
     // JSDOM doesn't implement form submissions
     // see https://github.com/jsdom/jsdom/issues/1937#issuecomment-321575590
     //
-    // If a form submission is caused by a input/button[type=submit], this instead outputs a
+    // If a form submission is caused by a input/button, this instead outputs a
```
Comment (tombye) [inline tests/javascripts/preventDuplicateFormSubmissions.test.js:66]: This, and the other replacement in this file, would read better as:
```suggestion
    // If a form submission is caused by a submit input/button, this instead outputs a
```

Also: could we update the HTML fixture under `// set up DOM` too?

File: tests/javascripts/previewPane.test.js [hunk 2]
```diff
@@ -137,7 +137,7 @@ describe('Preview pane', () => {
         // run preview pane script
         require('../../app/assets/javascripts/previewPane.js');
 
-        expect(document.querySelector('button[type=submit]').textContent).toEqual('Save');
+        expect(document.querySelector('form button').textContent).toEqual('Save');
```
Comment (tombye) [inline tests/javascripts/previewPane.test.js:140]: Can we update the HTML fixture under `// set up DOM` so the button doesn't have the `type="submit"` any more?

**Review by tombye — CHANGES_REQUESTED** [review]
Only a few, fairly petty requests for change. Overall looks good and I've tested across the browsers with these variants:
- buttons to submit forms sending multiple values (sign in page)
- buttons to submit forms sending single value (settings page)
- buttons to submit in-page search term (notifications page)
- buttons to upload file (add email branding page)

All looked good and worked well so, if the changes are done 👍🏻

**Review by leohemsted — COMMENTED** [review]
These changes seem pretty reasonable.

what i wouldn't mind, however, is [after this story] us having some thought to figure out how we can avoid swathes of HTML in our javascript and also in our test suites. Be good if we can call govuk-frontend stuff from there and not have to worry when we bump govuk that we might need to update all these things again

File: app/templates/components/cookie-banner.html [hunk 3]
```diff
@@ -5,10 +5,10 @@
     <h2 class="notify-cookie-banner__heading govuk-heading-m" id="notify-cookie-banner__heading">Can we store analytics cookies on your device?</h2>
     <p class="govuk-body">Analytics cookies help us understand how our website is being used.</p>
     <div class="notify-cookie-banner__buttons">
-      <button class="govuk-button notify-cookie-banner__button notify-cookie-banner__button-accept" type="submit" data-accept-cookies="true" aria-describedby="notify-cookie-banner__heading">
+      <button type="button" class="govuk-button notify-cookie-banner__button notify-cookie-banner__button-accept" type="button" data-accept-cookies="true" aria-describedby="notify-cookie-banner__heading">
```
[the remark this answers is not in the document]
Comment (tombye) [reply app/templates/components/cookie-banner.html]: Should be resolved by https://github.com/alphagov/notifications-admin/pull/4435/commits/a3f6a0f280a6ec76b8568be39c2e5ef1de323c4c

**Review by tombye — APPROVED** [review]
Since my last review, we found a few things that needed updating. I'm now confident all the buttons are up-to-date and using the right macro so 👍🏻

**Review by samuelhwilliams — APPROVED** [review]
Looks good to me 👀 👍  Nice work. All the things I spotted going commit-by-commit got resolved in later commits.

**Pull request merged**